core/src/main/scala/akka/persistence/dynamodb/internal/BySliceQuery.scala

```diff
@@ -300,7 +300,7 @@ import org.slf4j.Logger
     }
 
     def nextQuery(state: QueryState): (QueryState, Option[Source[Envelope, NotUsed]]) = {
-      val newIdleCount = if (state.itemCount == 0) state.idleCount + 1 else 0
+      val newIdleCount = if (state.itemCount == 0 && !state.backtracking) state.idleCount + 1 else 0
```

Comment:
Since the backtracking query has a definite endpoint in the past, it's not unexpected that a query would return zero items nor would returning zero items suggest anything about whether we're caught up.